pkg/util/test/util.go

```diff
@@ -3017,88 +3044,231 @@ func ValidateTelemetry(pxImageList map[string]string, cluster *corev1.StorageClu
+		} else if host == "" || port == "" {
+			return "", "", "", fmt.Errorf("failed to split px proxy to get host and port [%s]", proxy)
+		}
+		return host, port, authHeader, nil
 	}
+}
 
-	logrus.Debugf("Sleeping for 2 minutes to get new Telemetry state")
-	time.Sleep(2 * time.Minute)
-
-	// Get StorageCluster
-	newCluster, err := operatorops.Instance().GetStorageCluster(liveCluster.Name, liveCluster.Namespace)
-	if err != nil {
-		return nil, err
+// CanAccessArcusRegisterEndpoint checks if telemetry registration endpoint is reachable
+// return true immediately if it can reach to Arcus
+// return false after failing 5 times in a row
+func CanAccessArcusRegisterEndpoint(
+	cluster *corev1.StorageCluster,
+	httpProxy string,
+) bool {
+	endpoint := getArcusRegisterProxyURL(cluster)
+	logrus.Debugf("Checking whether telemetry registration endpoint [%s] is accessible on cluster [%s]...",
+		endpoint, cluster.Name)
+
+	url, _ := url.Parse(fmt.Sprintf("https://%s:443/auth/1.0/ping", endpoint))
+	request := &http.Request{
+		Method: "GET",
+		URL:    url,
+		Header: map[string][]string{
+			// 3 headers are required here by the API, but can use dummy values here.
+			// cluster UUID can be empty, so not using it for appliance-id here.
+			"product-name": {"portworx"},
+			"appliance-id": {"portworx"},
+			"component-sn": {cluster.Name},
+		},
+	}
+	client := &http.Client{}
+	if httpProxy != "" {
+		if !strings.HasPrefix(strings.ToLower(httpProxy), "http://") {
+			httpProxy = "http://" + httpProxy
+		}
+		proxyURL, err := url.Parse(httpProxy)
+		if err != nil {
+			logrus.Errorf("failed to parse http proxy [%s] for checking Pure1 connectivity, Err: %v", httpProxy, err)
+			return false
+		}
+		client.Transport = &http.Transport{
+			Proxy: http.ProxyURL(proxyURL),
+		}
+	}
+	for i := 1; i <= arcusPingRetry; i++ {
+		response, err := client.Do(request)
+		warnMsg := fmt.Sprintf("Failed to access telemetry registration endpoint [%s]", endpoint)
+		if err != nil {
+			logrus.WithError(err).Warnf(warnMsg)
+		} else if response.StatusCode != 200 {
+			// Only consider 200 as a successful ping with a properly constructed request.
+			body, _ := io.ReadAll(response.Body)
+			response.Body.Close()
+			logrus.WithFields(logrus.Fields{
+				"code": response.StatusCode,
+				"body": string(body),
+			}).Warnf(warnMsg)
+		} else {
+			logrus.Infof("Telemetry registration endpoint [%s] is accessible on cluster [%s]",
+				endpoint, cluster.Name)
+			return true
+		}
+		if i != arcusPingRetry {
+			logrus.Debugf("Failed to ping Pure1 [%s], retrying...", endpoint)
+			time.Sleep(arcusPingInterval)
+		}
 	}
+	return false
+}
 
-	// Update Telemetry status
-	if cluster.Spec.Monitoring != nil && cluster.Spec.Monitoring.Telemetry != nil {
-		cluster.Spec.Monitoring.Telemetry.Enabled = newCluster.Spec.Monitoring.Telemetry.Enabled
-	} else {
-		cluster.Spec.Monitoring = newCluster.Spec.Monitoring
+func getArcusTelemetryLocation(cluster *corev1.StorageCluster) string {
```

Comment:
This function will make our default location to be external(prod) unless specified in the annotation. Hope that's the expectation here

Reply:
It is expected, this is a general validation for both production and staging telemetry